fix: use correct symbol for referencing branches on forges

H1ghBre4k3r opened the pull request:
## 🧢 Changes

Use the correct symbol for referencing branches on different forges. 

## ☕️ Reasoning

The issue arose when using stacked branches on GitLab in a merge request. We used `#` instead of `!` which worked fine for GitHub but did not work  for GitLab. 

Now, we utilize the already present information in the `ForgePrService`  such that we always use the correct symbol when referencing branches.

Caleb-T-Owens commented on apps/desktop/src/lib/forge/shared/prFooter.ts:
```diff
@@ -33,18 +33,23 @@ export async function updatePrDescriptionTables(prService: ForgePrService, prNum
  * Replaces or inserts a new footer into an existing body of text.
  */
-function updateBody(body: string | undefined, prNumber: number, allPrNumbers: number[]) {
+function updateBody(
```
Could we pass the symbol in here too, so that only `updatePrDescriptionTables` needs a ForgeService in order to be tested?

Caleb-T-Owens reviewed (approved):
LGTM. Fire at will

H1ghBre4k3r committed:
fix: use correct symbol for referencing branches on forges

Use the correct symbol for referencing branches on different forges. The
issue arose when using stacked branches on GitLab in a merge request. We
used `#` instead of `!` which worked fine for GitHub but did not work 
for GitLab. 

Now, we utilize the already present information in the `ForgePrService` 
such that we always use the correct symbol when referencing branches.
